runtime/coverage.go

```diff
@@ -108,23 +107,23 @@ func NewLocationCoverage(lineHits map[int]int) *LocationCoverage {
 // also exclude locations from coverage collection.
 type CoverageReport struct {
 	// Contains a *LocationCoverage per location.
-	Coverage map[common.Location]*LocationCoverage `json:"-"`
+	Coverage map[string]*LocationCoverage `json:"-"`
```

Comment:
Note: I have changed the type here, from `common.Location` to `string`, in order to facilitate the JSON unmarshal. I am not sure if there's any method available, that can convert a string to its respective `common.Location` implementation, e.g `AddressLocation`, `ScriptLocation`, `TransactionLocation` or `StringLocation`. The public API still accepts `common.Location` arguments, and uses the newly-added `ID` method, to convert it to a string.

Reply:
Thanks for pointing this out :clap: I was not aware of it.
My main concern is that the `ID` method truncates some information from locations (e.g qualified identifiers), which means that we are not able to map it directly with a `common.Location` implementation. To instantiate them, there are checks in place, such as:

```go
pieceCount := len(parts)
if pieceCount == 1 {
	return newError("missing qualified identifier")
}
```

which prevent us from creating the proper `common.Location` implementation, from its `ID`. Is there any workaround to this? Should we update the `ID` method to also include any qualified identifiers, thus enabling us to instantiate it afterwards, by the respective decoder (e.g `decodeFlowLocationTypeID`)?

Reply:
As a starting point, I have added a first implementation on https://github.com/onflow/cadence/pull/2391/commits/e9110b9f1041e300481361f948570c47ac711e3e, so we have something to refine. I am open to better alternatives :bow:

Reply:
Ah, you are right, I didn't think about the decoding part.

Basically, so far we had a type ID, which is location + qualified identifier. Type IDs can be decoded using `DecodeTypeID`. 

Now we added a location ID, and we also want a decoding function `DecodeLocation`.

Thanks for getting started on an implementation, I'll have a look 👌 We might want to move it to `common` (maybe as `DecodeLocationID`?), and generalize it like in `DecodeTypeID`, so that we can add further location kinds in the future.

Reply:
I'll open a separate PR to rework `DecodeTypeID` to make the missing qualified identifier part a non-error, so the function can be used to also just decode location IDs.

Reply:
That would be great :pray:

Reply:
Opened #2396

Reply:
PR is rebased on master, after https://github.com/onflow/cadence/pull/2396 was merged.